front/pages/w/[wId]/welcome.tsx

```diff
@@ -0,0 +1,171 @@
+import { Button, Input, RadioButton } from "@dust-tt/sparkle";
+import { GetServerSideProps, InferGetServerSidePropsType } from "next";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+import { getUserMetadata } from "@app/lib/api/user";
+import { Authenticator, getSession, getUserFromSession } from "@app/lib/auth";
+import { UserType, WorkspaceType } from "@app/types/user";
+
+const { URL = "", GA_TRACKING_ID = "" } = process.env;
+
+export const getServerSideProps: GetServerSideProps<{
+  user: UserType;
+  owner: WorkspaceType;
+  defaultExpertise: string;
+  conversationId: string | null;
+  gaTrackingId: string;
+  baseUrl: string;
+}> = async (context) => {
+  const session = await getSession(context.req, context.res);
+  const user = await getUserFromSession(session);
+  const auth = await Authenticator.fromSession(
+    session,
+    context.params?.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner || !auth.isUser() || !user) {
+    return {
+      redirect: {
+        destination: "/",
+        permanent: false,
+      },
+    };
+  }
+
+  const expertise = await getUserMetadata(user, "expertise");
+
+  // If user was in onboarding flow "domain_conversation_link"
+  // We will redirect to the conversation page after onboarding.
+  const conversationId =
+    typeof context.query.cId === "string" ? context.query.cId : null;
+
+  return {
+    props: {
+      user,
+      owner,
+      defaultExpertise: expertise?.value || "",
+      conversationId,
+      baseUrl: URL,
+      gaTrackingId: GA_TRACKING_ID,
+    },
+  };
+};
+
+export default function Welcome({
+  user,
+  owner,
+  defaultExpertise,
+  conversationId,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const router = useRouter();
+  const [firstName, setFirstName] = useState<string>(user.name.split(" ")[0]);
+  const [lastName, setLastName] = useState<string>(user.name.split(" ")[1]);
+  const [expertise, setExpertise] = useState<string>(defaultExpertise);
+  const [isFormValid, setIsFormValid] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsFormValid(
+      firstName.length > 0 && lastName.length > 0 && expertise !== ""
+    );
+  }, [firstName, lastName, expertise]);
+
+  const handleSubmit = async () => {
+    const updateUserFullNameRes = await fetch("/api/user", {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ firstName, lastName }),
+    });
+    if (updateUserFullNameRes.ok) {
+      await fetch("/api/user/metadata/expertise", {
```

Comment:
UserMetadata are meant to be deletable at will without breaking the product (think cookie). Let's not use that for this if we need to use it at a later stage :+1: